pkg/network/encoding/marshal/usm.go

```diff
@@ -97,6 +103,26 @@ func (bc *USMConnectionIndex[K, V]) Find(c network.ConnectionStats) *USMConnecti
 	return result
 }
 
+func (bc *USMConnectionIndex[K, V]) find(c network.ConnectionStats) *USMConnectionData[K, V] {
+	result := bc.lookupFn(c, bc.data)
+	if result != nil || !bc.enableConnectionRollup {
+		return result
+	}
+
+	// If no matches were found and we have connection rollups enabled, perfom
+	// 2 additional lookups with each port set to 0
+	tmp := c.SPort
+	c.SPort = 0
+	result = bc.lookupFn(c, bc.data)
+	if result != nil {
+		return result
+	}
+
+	c.SPort = tmp
+	c.DPort = 0
+	return bc.lookupFn(c, bc.data)
```

Comment:
Can't that generate false positives?
For example having
c1: `127.0.0.1:12345 <-> 127.0.0.1:80`
c2: `127.0.0.1:12345 <-> 127.0.0.1:81`

and we'll aggregate the them using: `127.0.0.1:12345 <-> 127.0.0.1:*`?

Reply:
This part of the code will only perform lookups, so I'm guessing you're asking about the rollup algorithm itself?
If we have 
``` 
c1: 127.0.0.1:12345 <-> 127.0.0.1:80
c2: 127.0.0.1:12345 <-> 127.0.0.1:81
```
Then yes, the algorithm will consider the "right side" to the be the client side and will produce the following aggregation:
```
127.0.0.1:12345 <-> 127.0.0.1:0
```
Despite the fact that that 80 and 81 are outside of the ephemeral range and therefore make the example confusing, I think that's the correct behavior? If the client side is `127.0.0.1:12345`, how would that source port hit 2 distinct destinations? In order for that to happen you would need a port collision(re-use) in a 30second window AND for that to hit a different remote address, which to me seems like a very non-sensical/far-fetched scenario.

Reply:
I'm guessing this is more of an edge case than the standard behavior